command/registry.go

```diff
@@ -0,0 +1,176 @@
+package command
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/packer/hcl2template"
+	"github.com/hashicorp/packer/internal/registry"
+	"github.com/hashicorp/packer/internal/registry/env"
+	"github.com/hashicorp/packer/packer"
+
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/gocty"
+)
+
+const (
+	// Known HCP Packer Image Datasource, whose id is the SourceImageId for some build.
+	hcpImageDatasouceType string = "hcp-packer-image"
+	buildLabel            string = "build"
+)
+
+// SetupRegistry attempts to setup the HCP-related structures if HCP is enabled
+// for the command
+func SetupRegistry(cfg packer.Handler) hcl.Diagnostics {
+	switch cfg := cfg.(type) {
+	case *hcl2template.PackerConfig:
+		return setupRegistryForPackerConfig(cfg)
+	case *CoreWrapper:
+		return setupRegistryForPackerCore(cfg)
+	}
+
+	panic("NOPE.")
```

Comment:
Probably want a more descriptive error here and instead of a panic we what to return a hcl.DiagError

Reply:
Oh yeah, that's a remnant from my local dev tests :sweat_smile: 
I'll change this